kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/Checkpointer.java

```diff
@@ -98,9 +99,18 @@ public Optional<CheckpointMetaData> readLastCheckpointFile(TableClient tableClie
      */
     private Optional<CheckpointMetaData> loadMetadataFromFile(TableClient tableClient) {
         try {
-            // For now we use file size = 0 and modification time = 0, in the future we should use
-            // listFrom to retrieve the real values see delta-io/delta#2140
-            FileStatus lastCheckpointFile = FileStatus.of(lastCheckpointFilePath.toString(), 0, 0);
+            FileSystemClient fileSystemClient = tableClient.getFileSystemClient();
+            FileStatus lastCheckpointFile = null;
+            try (CloseableIterator<FileStatus> lastCheckpointFileStatusIterator =
+                fileSystemClient.listFrom(
+                    this.lastCheckpointFilePath.toString()
+                )
+            ) {
+                while (lastCheckpointFileStatusIterator.hasNext()) {
+                    lastCheckpointFile = lastCheckpointFileStatusIterator.next();
+                }
+            }
+
             JsonHandler jsonHandler = tableClient.getJsonHandler();
```

Comment:
Need to handle the case when `lastCheckpointFile` doesn't exists, basically return `Optional.empty`.

Reply:
I believe this is already handled in the catch block on line 129, but do correct me if I am wrong